keps/1284-cluster-queue-stop/README.md

```diff
@@ -0,0 +1,131 @@
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+Add setting in a ClusterQueue that an administrator is able to use in order to to pause new admissions and have the option to cancel current QuotaReservations and Evict admitted workloads.
+
+## Motivation
+
+This is a common admin journey to control usage from a user.
+
+### Goals
+
+Add a setting in a ClusterQueue that an administrator is able to use in order to to pause new admissions and have the option to cancel current QuotaReservations and Evict admitted workloads.
+
+### Non-Goals
+
+Manage the QuotaReservation and Admission of workloads form the same cohort that might borrow resources from the ClusterQueue in question.
+
+## Proposal
+
+Add a new member in the ClusterQueue implementation `stopPolicy` the presence of which will mark the ClusterQueue as Inactive and it's value will control how the `Admitted` or `Reserving` workloads are affected.
+
+### User Stories
+#### Story 1
+
+As a cluster administrator I want to be able to stop the new admissions in a specific ClusterQueue with the option of Evicting currently admitted Workloads or canceling QuotaReservations.
+
+### Risks and Mitigations
+
+Managing the Reservation canceling and Eviction of workloads in other queues from the same cohort that
+are potentially borrowing resources form the stopped queue adds a considerable amount of complexity
+while having a limited added value, therefore these cases are not covered in this first iteration. 
+
+## Design Details
+
+### API/ClusterQueue
+
+```go
+type ClusterQueueSpec struct {
+	// ....
+
+	// stopPolicy - if set the ClusterQueue is considered Inactive, no new reservation being
+	// made. 
+	//
+	// Depending on its value, its associated workloads will:
+	//
+	// - StopNow - Admitted workloads are evictred and Reserving workloads will cancel the reservatio.
+	// - WaitForAdmitted - Admitted workloads will run to completion and Reserving workloads will cancel the reservatio.
+	// - WaitForReserving - Admitted and Reserving workloads will run to completion.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=StopNow;WaitForAdmitted;WaitForReserving
+	StopPolicy StopPolicy `json:"stopPolicy,omitempty"`
+}
+
+type StopPolicy string
+
+const (
+	StopNow StopPolicy = "StopNow"
+	WaitForAdmitted StopPolicy = "WaitForAdmitted"
+	WaitForReserving StopPolicy = "WaitForReserving"
+)
+
+
+```
+### Controllers
+#### ClusterQueue
+
+Once the `stopPolicy` is set the cluster queue is marked as inactive with a relevant status message.
+
+#### Workload
+
+If the cluster queue associated to a workload has the `stopPolicy` changed depending on the policy value and state of the
```

Comment:
So whenever a CQ gets the stopPolicy updated, we'll trigger reconciliation for every workload in the queue, correct?

Reply:
correct